odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -318,6 +318,9 @@ def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
         self._update_reference_fields(src_partners, dst_partner)
         self._update_values(src_partners, dst_partner)
 
+        self.env.add_to_compute(dst_partner._fields['partner_share'], dst_partner)
```

Comment:
Not against this specific change but normally we have a specific method to find what we need to recompute depending of the change done on the recordset: `modified([<field_change>])` (check write by example).